scipy/special/special/tools.h

```diff
@@ -0,0 +1,70 @@
+/* Building blocks for implementing special functions */
+
+#pragma once
+
+#include "config.h"
+#include "error.h"
+
+namespace special {
+namespace detail {
+
+    // Series evaluators.
+    template <typename Generator, typename Number>
```

Comment:
Do we really need to worry about Number being non-floating point? I see you have this constexpr if below. Is there a use case for this?

Reply:
We want this generic over both regular `floating point` and `complex`. The `constexpr` thing is just a cookbook recipe I found online for checking if its a real type or a complex type